crypto/src/main/kotlin/net/corda/v5/crypto/DigitalSignature.kt

```diff
@@ -36,4 +36,19 @@ open class DigitalSignature(
          */
         val context: Map<String, String>
     ) : DigitalSignature(bytes)
+
+    class WithMetadata(
+        /**
+         * Public key which corresponding private key was used to sign the data (as if an instance
+         * of the [CompositeKey] is passed to the sign operation it may contain keys which are not actually owned by
+         * the member).
+         */
+        by: PublicKey,
+        bytes: ByteArray,
+        /**
+         * The metadata which was used by the signing operation, note that the metadata **is signed over**.
+         * The format will be stable and consistent. The actual signing data is concatenation of the serialised metadata and the  payload data.
+         */
+        metadata: Map<String, String>,
```

Comment:
I do think the DigitalSignatureAndMetadata was clearer than this internal class and note that this isn't marked as CordaSerializable, so it isn't in fact going to work. We should also make the metadata SortedMap<String,String> so that the ordering is explicit (and perhaps to some degree automatic?)